querydsl-core/src/main/java/com/querydsl/core/DefaultQueryMetadata.java

```diff
@@ -59,7 +61,8 @@
 
     private List<OrderSpecifier<?>> orderBy = ImmutableList.of();
 
-    private List<Expression<?>> projection = ImmutableList.of();
+    @Nullable
```

Comment:
I think it's a better idea to pull this annotation up to the interface, if you want to get hints from tools.